client/Hooks/useLogger/useLogger.steps.tsx

```diff
@@ -0,0 +1,131 @@
+  logger = renderedHook.result.current;
+  expect(logger).not.toBeNull();
+  expect(logger).not.toBeUndefined();
+  rerenderHook = renderedHook.rerender;
+  expect(rerenderHook).not.toBeNull();
+  expect(rerenderHook).not.toBeUndefined();
+};
+When(useLoggerHookIsRendered, useLoggerHookIsRenderedFn);
+And(useLoggerHookIsRendered, useLoggerHookIsRenderedFn);
+
+And('the useLogger hook is re-rendered', () => {
+  rerenderHook();
+});
+
+When('the useLogger hook is rendered with an empty componentName', () => {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const renderedHook = renderHook(() => useLogger(''));
+  logger = renderedHook.result.current;
+  expect(logger).not.toBeNull();
+  expect(logger).not.toBeUndefined();
+});
+
+And(
+  /^(\d+) (fatal|error|warn|info|debug|trace)-level messages* (?:are|is) logged$/,
+  (messagesCount, logLevel) => {
+    const requestedMessagesCount =
+      parseInt(messagesCount as string) + sentMessagesCount;
+    for (
+      let index = sentMessagesCount + 1;
+      index <= requestedMessagesCount;
+      index++
+    ) {
+      logger[logLevel as string](`useLogger test message ${index}`);
+    }
+    sentMessagesCount = requestedMessagesCount;
+  }
+);
+
+const useLoggerHookIsConnected =
+  'the useLogger hook is connnected to the WebSocket server';
+const useLoggerHookIsConnectedFn = () => {
+  expect(mockWebSocketServer.clients().length).toBe(1);
+  return onConnectionPromise;
+};
+Then(useLoggerHookIsConnected, useLoggerHookIsConnectedFn);
+And(useLoggerHookIsConnected, useLoggerHookIsConnectedFn);
+
+Then('the useLogger hook does not connnect to the WebSocket server', () => {
+  expect(mockWebSocketServer.clients().length).toBe(0);
+});
+
+const loggingMessagesAreReceived = /^(\d+) (fatal|error|warn|info|debug|trace)-level logging messages* (?:are|is) received by the WebSocket server$/;
+const loggingMessagesAreReceivedFn = async (messagesCount, logLevel) => {
+  const expectedLoggingMessagesCount = parseInt(messagesCount as string);
+
+  const receivedMessageEvent = await onMessagePromise;
+  expect(receivedMessageEvent).not.toBeNull();
+  const parsedLoggingMessages = JSON.parse(receivedMessageEvent);
+  expect(parsedLoggingMessages.length).toBe(expectedLoggingMessagesCount);
+
+  let startingIndex = 1;
+  if (sentMessagesCount > MESSAGE_BUFFER_MAX_SIZE) {
+    expect(parsedLoggingMessages.length).toBe(MESSAGE_BUFFER_MAX_SIZE);
+    startingIndex = sentMessagesCount - MESSAGE_BUFFER_MAX_SIZE + 1;
+  }
+
+  for (let index = 0; index < expectedLoggingMessagesCount; index++) {
+    expect(parsedLoggingMessages[index].clientLevel).toEqual(logLevel);
+    expect(parsedLoggingMessages[index].componentName).toEqual('TestComponent');
+    expect(parsedLoggingMessages[index].msg).toEqual(
+      `useLogger test message ${index + startingIndex}`
+    );
+  }
+};
+Then(loggingMessagesAreReceived, loggingMessagesAreReceivedFn);
+And(loggingMessagesAreReceived, loggingMessagesAreReceivedFn);
```

Comment:
Eugh, that's unfortunate syntax :(

Reply:
Agreed, ugly. Couldn't spot any alternative way to use `And` for the same code as a `Given`, `When` or `Then`. If you have any good ideas... :)